jina/excepts.py

```diff
@@ -1,93 +1,97 @@
 
 
-class BadConfigSource(FileNotFoundError, BaseJinaExeception):
+class BadConfigSource(FileNotFoundError, BaseJinaException):
     """The yaml config file is bad, not loadable or not exist."""
 
 
-class BadClient(Exception, BaseJinaExeception):
+class BadClient(Exception, BaseJinaException):
     """A wrongly defined client, can not communicate with jina server correctly."""
 
 
-class BadClientCallback(BadClient, BaseJinaExeception):
+class BadClientCallback(BadClient, BaseJinaException):
     """Error in the callback function on the client side."""
 
 
-class BadClientInput(BadClient, BaseJinaExeception):
+class BadClientInput(BadClient, BaseJinaException):
     """Error in the request generator function on the client side."""
 
 
-class BadRequestType(TypeError, BaseJinaExeception):
+class BadRequestType(TypeError, BaseJinaException):
     """Exception when can not construct a request object from given data."""
 
 
-class BadImageNameError(Exception, BaseJinaExeception):
+class BadClientResponse(Exception, BaseJinaException):
+    """Exception when can not construct a request object from given data."""
+
+
+class BadImageNameError(Exception, BaseJinaException):
     """Exception when an image name can not be found either local & remote"""
 
 
-class BadYAMLVersion(Exception, BaseJinaExeception):
+class BadYAMLVersion(Exception, BaseJinaException):
     """Exception when YAML config specifies a wrong version number."""
 
 
-class DaemonConnectivityError(Exception, BaseJinaExeception):
+class DaemonConnectivityError(Exception, BaseJinaException):
     """Exception to raise when jina daemon is not reachable."""
 
 
-class DaemonWorkspaceCreationFailed(Exception, BaseJinaExeception):
+class DaemonWorkspaceCreationFailed(Exception, BaseJinaException):
     """Exception to raise when jina daemon is not connectable."""
 
 
-class DaemonPodCreationFailed(Exception, BaseJinaExeception):
+class DaemonPodCreationFailed(Exception, BaseJinaException):
     """Exception to raise when jina daemon is not connectable."""
 
 
-class NotSupportedError(Exception, BaseJinaExeception):
+class NotSupportedError(Exception, BaseJinaException):
     """Exception when user accidentally using a retired argument."""
 
 
-class RuntimeRunForeverEarlyError(Exception, BaseJinaExeception):
+class RuntimeRunForeverEarlyError(Exception, BaseJinaException):
     """Raised when an error occurs when starting the run_forever of Runtime"""
 
 
-class DockerVersionError(SystemError, BaseJinaExeception):
+class DockerVersionError(SystemError, BaseJinaException):
     """Raised when the docker version is incompatible"""
 
 
-class DaemonInvalidDockerfile(FileNotFoundError, BaseJinaExeception):
+class DaemonInvalidDockerfile(FileNotFoundError, BaseJinaException):
     """Raised when invalid dockerfile is passed to JinaD"""
 
 
-class NoContainerizedError(Exception, BaseJinaExeception):
+class NoContainerizedError(Exception, BaseJinaException):
```

Comment:
Are we sure? not sure if it is safe from multi inheritance p.o.v

Reply:
I added a deprecation warning so the old one is still usable